test/DAICO.test.js

```diff
@@ -33,85 +33,188 @@ contract('DAICO', function (accounts) {
+    it('if you did not purchased a token you are not a holder', async function() {
       let purchaser = accounts[4];
       await this.daico.buyTokens(purchaser, {from: purchaser, value: higher_value}).should.be.fulfilled;
       let isholder = await this.daico.isHolder(purchaser);
       assert.ok(isholder);
     });
 
-    it('holder can propose a new tap', async function() {
+    it('holder can propose a new tap in tap mode', async function() {
       let purchaser = accounts[4];
       await this.daico.buyTokens(purchaser, {from: purchaser, value: higher_value}).should.be.fulfilled;
+      await increaseTimeTo(this.afterlastWithdrawn);
       await this.daico.newRaiseTapProposal(tap * 2, 3600, {from: purchaser}).should.be.fulfilled;
     });
 
     it('if you are not a holder you cant propose a new tap', async function() {
       let noholder = accounts[4];
+      await increaseTimeTo(this.afterlastWithdrawn);
       await this.daico.newRaiseTapProposal(tap * 2, 3600, {from: noholder}).should.be.rejectedWith(EVMRevert);
     });
 
-    it('you can not propose lower the tap', async function() {
+    it('holder can vote a proposal in time', async function() {
       let purchaser = accounts[4];
-      await this.daico.buyTokens(purchaser, {from: purchaser, value: higher_value}).should.be.fulfilled;
-      await this.daico.newRaiseTapProposal(100, 3600, {from: purchaser}).should.be.rejectedWith(EVMRevert);
+      await this.daico.buyTokens(purchaser, {from: purchaser, value: higher_value});
+      await increaseTimeTo(this.afterlastWithdrawn);
+      await this.daico.newRaiseTapProposal(100, 3600, {from: purchaser}).should.be.fulfilled;
+      await this.daico.vote(0, true, {from: purchaser}).should.be.fulfilled;
+    });
+
+    it('no one can vote a proposal after votingDeadline', async function() {
+      let purchaser = accounts[4];
+      await this.daico.buyTokens(purchaser, {from: purchaser, value: higher_value});
+      await increaseTimeTo(this.afterlastWithdrawn);
+      const proposalID = await this.daico.newRaiseTapProposal(100, 3600, {from: purchaser});
+      await increaseTimeTo(latestTime() + duration.hours(1));
+      await this.daico.vote(0, true, {from: purchaser}).should.be.rejectedWith(EVMRevert);
     });
 
+    it('passed propasal can be executed', async function() {
+      let purchaser = accounts[4];
+      await this.daico.buyTokens(purchaser, {from: purchaser, value: higher_value});
+      await increaseTimeTo(this.afterlastWithdrawn);
+      const proposalID = await this.daico.newRaiseTapProposal(100, 3600, {from: purchaser});
+      await this.daico.vote(0, true, {from: purchaser}).should.be.fulfilled;;
+      await increaseTimeTo(latestTime() + duration.hours(1) + duration.seconds(1));
+      let old_tap = await this.daico.tap();
+      assert.ok(old_tap.c[0]==0);
+      await this.daico.executeRaiseTapProposal(0);
+      let new_tap = await this.daico.tap();
+      assert.ok(new_tap.c[0]==100);
```

Comment:
we should check the proposal has been updated as we expected

Reply:
I did this in `00663f0` commit